### PR TITLE
[SRCH-995] Update Javascript API library to include CFAR types

### PR DESCRIPTION
**Pull request by matthewp5**
## 🔍 Overview

This PR updates the `OfferAvailableService` interface to support the CFAR service. 

## 🧑‍💻 What has been done?

- Added `OfferAvailableServiceCFAR` and `OfferAvailableServiceCFARMetadata` interfaces to offer types
- Moved existing baggage service type into `OfferAvailableServiceBaggage` interface
- Updated `OfferAvailableService` interface to be discriminated union using above types

### 🛠️ Other changes

- Update tests
- Updated mocks (and fixed non-matching passenger IDs)

## 🦮 Reviewer guide

I'd recommend reviewing this PR commit-by-commit.

### CODE REVIEW

File: src/booking/Offers/OfferTypes.ts [hunk 1]
```diff
@@ -164,82 +164,123 @@ export interface OfferAvailableServiceBaggageMetadata {
...
+   * This will be null only for orders where `awaiting_payment` is `false`.
+   */
+  payment_required_by?: string | null
+  /**
+   *  The ISO 8601 datetime at which the price associated
+   * with the order will no longer be guaranteed by the airline
+   * and the order will need to be repriced before payment.
+   * This can be null when there is no price guarantee.
+   */
+  price_guarantee_expires_at?: string | null
```
Comment (go1t) [inline src/booking/Offers/OfferTypes.ts:279]: nit: we probably don't need these `?`s right? I think we can guarantee that these fields will be included in the response.

File: src/booking/Offers/OfferTypes.ts [hunk 2]
```diff
@@ -164,82 +164,123 @@ export interface OfferAvailableServiceBaggageMetadata {
...
    */
-  type: T_ServiceType
+  type: 'cancel_for_any_reason'
+}
+
+export type OfferAvailableService =
+  | OfferAvailableServiceBaggage
+  | OfferAvailableServiceCFAR
+
+export interface PaymentRequirements {
```
Comment (matthewp5) [inline src/booking/Offers/OfferTypes.ts:265]: Note: I haven't added this, just moved it below all the service types.

**Review by go1t — APPROVED** [review]
LGTM

File: src/booking/Offers/OfferTypes.ts (hunk 1 above)
Comment (go1t) [reply src/booking/Offers/OfferTypes.ts:279]: oh I just saw your above comment. Feel free to leave this for a separate PR if you want!

Comment (matthewp5) [reply src/booking/Offers/OfferTypes.ts]: Done!

**Review by go1t — APPROVED** [review]
Thank you!

**Pull request merged**